### PR TITLE
Ci fixes

### PR DESCRIPTION
**Pull request by pepeiborra**
See commit messages

### CODE REVIEW

**Comment by jneira** [comment]
thanks, I hope we can recover reasonable ci build times with those nice changed

**Review by jneira — APPROVED** [review]

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -30,14 +30,16 @@ jobs:
             ghc-lib: true
 
     steps:
+      # Cancel queued workflows from earlier commits in this branch
+      - uses: fkirc/skip-duplicate-actions@master
```
Comment (jneira) [inline .github/workflows/test.yml:34]: great move

**Comment by jneira** [comment]
What about restrict tests to last minor versions, like we did in ghcide? We could include macos and still save jobs

**Comment by jneira** [comment]
Where those compile error comes from? weird thing that changing ci settings could trigger them

**Comment by pepeiborra** [comment]
I don't know if the auto cancel action will work - if it doesn't we'll need to keep an eye on https://github.community/t/github-actions-cancel-redundant-builds-not-solved/16025

**Comment by jneira** [comment]
@pepeiborra i've triggered two [consecutive](https://github.com/haskell/haskell-language-server/runs/1634696888) builds and the cancel action has failed with

```
Warning: HttpError: Resource not accessible by integration
Warning: Failed to cancel https://github.com/haskell/haskell-language-server/actions/runs/456673917
Do not skip execution because we did not find a transferable run
```

otoh i've observed the build times of cabal are significative better in the second build, at least for ubuntu:
* 25m 25s https://github.com/haskell/haskell-language-server/runs/1634643729?check_suite_focus=true
* 3m 33s https://github.com/haskell/haskell-language-server/runs/1634696911?check_suite_focus=true

**Pull request merged**